### PR TITLE
Bugfix/security 479

### PR DESCRIPTION
**Pull request by fengxx**
follow up on SECURITY-479, call  
```
context=ApprovalContext.create().withCurrentUser().withItemAsKey(item)
ScriptApproval.get().configuring(scriptText, GroovyLanguage.get(), context);
 to check approval
```

### CODE REVIEW

**Comment by fengxx** [comment]
@daniel-beck please help review

**Comment by fengxx** [comment]
Related to  https://github.com/jenkins-infra/backend-update-center2/pull/134

File: splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy [hunk 1]
```diff
@@ -36,25 +37,33 @@ public class UserActionDSL {
...
+                        SecureGroovyScript script = new SecureGroovyScript(groovyScriptText, false, null).configuringWithKeyItem();;
                         script.evaluate(cl, binding)
                     } else {
                         //had to call setDelegate
-                        CompilerConfiguration cc = new CompilerConfiguration();
+                        CompilerConfiguration cc = GroovySandbox.createSecureCompilerConfiguration();
                         cc.scriptBaseClass = ClosureScript.class.name;
                         ImportCustomizer ic = new ImportCustomizer()
                         ic.addStaticStars(LogEventHelper.class.name)
                         ic.addStarImport("jenkins.model")
```
Comment (jglick) [inline splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy]: Generally such classes are not going to be accessible from the sandbox.

File: splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy [hunk 2]
```diff
@@ -36,25 +37,33 @@ public class UserActionDSL {
...
                 binding.setVariable("splunkins", delegate);
-                ClassLoader cl = Jenkins.getActiveInstance().getPluginManager().uberClassLoader;
+                binding.setVariable("build", build);
+                binding.setVariable("env", buildParameters);
                 // the script written before SecureGroovyScript was introduced, and can not be migrated automatically
                 boolean legacyScript = SplunkJenkinsInstallation.get().isLegacyMode();
                 try {
+                    ApprovalContext context = ApprovalContext.create().withCurrentUser().withItemAsKey(build.getParent());
+                    //saving pending for approval
+                    String groovyScriptText = ScriptApproval.get().configuring(scriptText, GroovyLanguage.get(), context);
```
Comment (jglick) [inline splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy]: `configuring` should be called when a script is being configured, not when it is being run.

File: splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy [hunk 3]
```diff
@@ -36,25 +37,33 @@ public class UserActionDSL {
...
+                    //check approval, will throw UnapprovedUsageException
+                    ScriptApproval.get().using(groovyScriptText, GroovyLanguage.get());
```
Comment (jglick) [inline splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy]: And neither `configuring` nor `using` should be called when using sandbox mode.

File: splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy (hunk 2 above)
Comment (fengxx) [inline splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy]: Script maybe loaded from a file stored on Jenkins server, the script path is configured by user, but the script content will change after configuration.

File: splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy (hunk 1 above)
Comment (fengxx) [inline splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy]: I just want to use whole script approval mode with Delegate enabled.

File: splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy (hunk 3 above)
Comment (fengxx) [inline splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy]: about "this mode can only work where there is a fixed script associated with a Job and configured by a clearly identified person", I think whole script mode is just store a hash in scriptApproval.xml which doesn't bind to Job or Person. When the hashes is not found, just store as PendingScript which doesn't use the context(Job or Person) for hashcode/equal method either, (equal is checked on script and lang only)

Comment (fengxx) [inline splunk-devops/src/main/groovy/com/splunk/splunkjenkins/UserActionDSL.groovy]: amended commit to remove sandbox reference, it should be using whole script mode

**Comment by daniel-beck** [comment]
@jglick PT another L

**Comment by fengxx** [comment]
sorry, what does` PT another L` mean?

**Comment by daniel-beck** [comment]
@fengxx *Please take another look*, I asked @jglick to re-review this.

**Comment by fengxx** [comment]
replaced by #3

**Pull request closed without merging**